src/assets/javascript/analytics/cookieBanner.js

```diff
@@ -0,0 +1,95 @@
+/* global document window */
+
+window.DI = window.DI || {};
+
+(function(DI) {
+
+  'use strict'
+
+  const COOKIES_PREFERENCES_SET = "cookies_preferences_set";
+  const cookiesAccepted = document.querySelector("#cookies-accepted");
+  const cookiesRejected = document.querySelector("#cookies-rejected");
+  const hideCookieBanner = document.querySelectorAll(".cookie-hide-button");
+  const cookieBannerContainer = document.querySelector(".govuk-cookie-banner");
+  const cookieBanner = document.querySelector("#cookies-banner-main");
+  const acceptCookies = document.querySelector('button[name="cookiesAccept"]');
+  const rejectCookies = document.querySelector('button[name="cookiesReject"]');
+
+  function cookieBannerInit(domain) {
+
+    if (typeof domain !== "string") {
+      /* eslint-disable-next-line no-console */
+      console.warn(`Cookie banner cannot be initialised: domain is ${domain}`)
+      return
+    }
+
+    acceptCookies.addEventListener(
+      "click",
+      function (event) {
+        event.preventDefault();
+        setBannerCookieConsent(true, domain);
+      }.bind(this)
```

Comment:
Why do we need to bind `this`?

Reply:
I think this is mainly convention - as I understand it we are binding the addEventListener handler function to the same 'this' context as outside the function, i.e. the window object. Not sure why it is being done specifically here as we don't refer to 'this' inside the handler. Removing it had no effect.

Reply:
Cool, I agree that if we're not using `this` within the function there is probably no reason to bind it. I would prefer that we remove it as it adds a little complexity to something that is already a bit of a maze to follow but at the end of the day it's not much code - I'll leave the decision to you